zetaclient/bitcoin_client.go

```diff
@@ -763,13 +827,22 @@ func (ob *BitcoinChainClient) SetNextNonce() error {
 			continue
 		}
 		if ob.chain.ChainId == nonce.ChainId && bytes.Equal(tssKey.PubkeyInBytes, ob.Tss.Pubkey()) {
-			ob.nextNonce = int(nonce.NonceLow)
-			ob.ts.SetNextNonce(ob.nextNonce)
+			// find lowest nonce that had not been broadcasted
+			next2Broadcast := int(nonce.NonceLow)
+			for nc := nonce.NonceLow; nc < nonce.NonceHigh; nc++ {
+				if _, exist := ob.broadcastedTx[ob.GetTxID(uint64(nc))]; exist {
+					next2Broadcast++
+					continue
+				}
+				break
+			}
+			ob.nextNonceBroadcast = next2Broadcast
+			ob.ts.SetNextNonce(ob.nextNonceBroadcast)
```

Comment:
By curiosity: why are there two locations where the next nonce is set in the same object? Wouldn't it make more sense to keep it in a single location?